crates/project/src/project.rs

```diff
@@ -8501,7 +8504,7 @@ impl Project {
                 path: PathBuf::from(serialized_symbol.path).into(),
             };
             let language = languages
-                .language_for_file(&path.path, None)
+                .language_for_file(&path.path, None, None)
```

Comment:
Why `None` is used here here and `Option` is used in general?
It feels that we have to load the language keeping the overrides in mind in every Zed corner?
I understand that some tests do not have any worktree initialized, but that alone does not seem like a great justification.

Same question for `semantic_index.rs`: that has `None` passed which means it will ignore user settings and index the project the way it wants? Looks bad since semantic search would not be able to produce proper results.

Reply:
That's why: https://github.com/zed-industries/zed/pull/7344#issuecomment-1925506168

Reply:
Yes, we do need AppContext anywhere to get dynamic settings, I will remove the Option wrapper.